src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -771,6 +771,37 @@ private BlockHeader[] FindHeadersReversedFull(BlockHeader startHeader, int numbe
             return result;
         }
 
+        public BlockHeader FindLowestCommonAncestor(BlockHeader firstDescendant, BlockHeader secondDescendant)
+        {
```

Comment:
this may loop forever until genesis block - needs a lookup limit (max reorg length)